libraries/botbuilder-dialogs/tests/skillDialog.test.js

```diff
@@ -282,23 +289,31 @@ describe('SkillDialog', function() {
                 skill: {},
                 skillHostEndpoint: 'http://localhost:3980/api/messages'
             } , 'SkillDialog');
+            dialog.state = {};
             
+            const conversationState = new ConversationState(new MemoryStorage());
+            const dialogState = conversationState.createProperty('dialogState');
+            const dialogs = new DialogSet(dialogState);
+            dialogs.add(dialog);
+            const dc = await dialogs.createContext(context);
+            dc.stack = [dialog];
             try {
-                await dialog.sendToSkill(context, {});
+                await dialog.beginDialog(dc, { activity });
             } catch (e) {
                 strictEqual(e.message, 'Whoops');
             }
         });
 
-        it('should not rethrow if Error.message is "Not Implemented"', (done) => {
+        it('should not rethrow if Error.message is "Not Implemented"', async () => {
             const adapter = new TestAdapter(/* logic param not required */);
-            const context = new TurnContext(adapter, { activity: {} });
+            const activity = { type: ActivityTypes.Message, channelId: Channels.Directline, conversation: { id: '1' } };
+            const context = new TurnContext(adapter, activity);
             context.turnState.set(adapter.OAuthScopeKey, DEFAULT_OAUTHSCOPE);
             const dialog = new SkillDialog({
                 botId: 'botId',
                 conversationIdFactory: { 
                     createSkillConversationIdWithOptions: async () => { throw new Error('Not Implemented'); },
-                    createSkillConversationId: async () => done()
+                    createSkillConversationId: async () => {}
                 },
                 conversationState: {
                     saveChanges: () => null
```

Comment:
These two tests were forced to change since `sendToSkill()` no longer calls `createSkillConversationId()`, which is what this tests. `beginDialog()` does call this and in order to switch to that, a lot of dialog state things needed to be created.